### ISSUE TITLE
关于用u2net来识别图像是否有object

### ISSUE THREAD

**Issue by foreverls** [NathanUA/U-2-Net#117]
最近刚刚开始学习object detection,谢谢大佬的开源！之前直接用u2net在自己的dataset上跑结果很好~可是如果加了完全是背景的图片结果就很奇怪， 所以自己试着想改成classification的问题，我把sigmoid换成了softmax, loss换成了cross entropy lossoutput channel也是变成了2，感觉这样是不是就把background 和 object当成了两个class, 但是如果background太多的话就会出现class imbalance issue? (对基础概念还不是很理解大佬见谅QAQ）不知道窝这么理解对不对惹~最后的结果目前来看不是很好， 不知道大佬有木有什么建议惹 打扰啦！😢

**Comment by NathanUA** [comment]
全黑背景确实会对结果有影响，而且不太好处理，你可以尝试加weights去balaance前后背景（基于像素的比例或者基于有无前景图片的比例来设置，但是这个取决于比例大小，如果比例不是很悬殊，这种基本没什么效果），也可以在train的时候丢掉完全背景的图片（如果这种图片不多的话）。softmax+cross-entropy 和 sigmoid+bce在二分类本质上是一样的，不会对结果有很大的影响。还有脏一点的方法就是先分类，再分割，或者直接作multi-task learning试试。
